### PR TITLE
Fixed debug compilation on Lassen

### LINKED ISSUE #1366: Develop doesn't build on Lassen with Clang in debug
```
ninja geosx
[37/41] Building CUDA object coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/CompositionalMultiphaseWell.cpp.o
FAILED: coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/CompositionalMultiphaseWell.cpp.o 
/usr/tce/packages/cuda/cuda-10.1.243/bin/nvcc -ccbin=/usr/tce/packages/clang/clang-upstream-2019.03.26/bin/clang++ -DCAMP_HAVE_CUDA -I/usr/WS2/corbett5/GEOSX/src/coreComponents -I/usr/WS2/corbett5/GEOSX/src/externalComponents -Iinclude -I/usr/WS2/corbett5/GEOSX/src/coreComponents/linearAlgebra -isystem=/usr/tce/packages/cuda/cuda-10.1.243/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/caliper/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/raja/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/pugixml/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include/conduit -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/hdf5/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/suitesparse/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/superlu_dist/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/trilinos/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/parmetis/include -isystem=/usr/tcetmp/packages/essl/essl-6.2.1/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/metis/include -restrict -arch sm_70 --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -g -G -O0 -Xcompiler -O0 -Xcompiler=-fPIC   -Xcompiler=-fopenmp=libomp -std=c++14 -x cu -c /usr/WS2/corbett5/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp -o coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/CompositionalMultiphaseWell.cpp.o && /usr/tce/packages/cuda/cuda-10.1.243/bin/nvcc -ccbin=/usr/tce/packages/clang/clang-upstream-2019.03.26/bin/clang++ -DCAMP_HAVE_CUDA -I/usr/WS2/corbett5/GEOSX/src/coreComponents -I/usr/WS2/corbett5/GEOSX/src/externalComponents -Iinclude -I/usr/WS2/corbett5/GEOSX/src/coreComponents/linearAlgebra -isystem=/usr/tce/packages/cuda/cuda-10.1.243/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/caliper/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/raja/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/pugixml/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include/conduit -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/hdf5/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/suitesparse/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/superlu_dist/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/trilinos/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/parmetis/include -isystem=/usr/tcetmp/packages/essl/essl-6.2.1/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/metis/include -restrict -arch sm_70 --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -g -G -O0 -Xcompiler -O0 -Xcompiler=-fPIC   -Xcompiler=-fopenmp=libomp -std=c++14 -x cu -M /usr/WS2/corbett5/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp -MT coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/CompositionalMultiphaseWell.cpp.o -o coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/wells/CompositionalMultiphaseWell.cpp.o.d
ptxas fatal   : Cannot take address of function '_ZNK5geosx12constitutive33MultiFluidPVTPackageWrapperUpdate7computeEddRKN7LvArray10ArraySliceIKdLi1ELi0ElEERKNS3_IdLi1ELi0ElEESA_SA_SA_RKNS3_IdLi2ELi1ElEERd' 
[38/41] Building CUDA object coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/CompositionalMultiphaseBase.cpp.o
FAILED: coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/CompositionalMultiphaseBase.cpp.o 
/usr/tce/packages/cuda/cuda-10.1.243/bin/nvcc -ccbin=/usr/tce/packages/clang/clang-upstream-2019.03.26/bin/clang++ -DCAMP_HAVE_CUDA -I/usr/WS2/corbett5/GEOSX/src/coreComponents -I/usr/WS2/corbett5/GEOSX/src/externalComponents -Iinclude -I/usr/WS2/corbett5/GEOSX/src/coreComponents/linearAlgebra -isystem=/usr/tce/packages/cuda/cuda-10.1.243/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/caliper/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/raja/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/pugixml/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include/conduit -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/hdf5/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/suitesparse/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/superlu_dist/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/trilinos/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/parmetis/include -isystem=/usr/tcetmp/packages/essl/essl-6.2.1/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/metis/include -restrict -arch sm_70 --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -g -G -O0 -Xcompiler -O0 -Xcompiler=-fPIC   -Xcompiler=-fopenmp=libomp -std=c++14 -x cu -c /usr/WS2/corbett5/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp -o coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/CompositionalMultiphaseBase.cpp.o && /usr/tce/packages/cuda/cuda-10.1.243/bin/nvcc -ccbin=/usr/tce/packages/clang/clang-upstream-2019.03.26/bin/clang++ -DCAMP_HAVE_CUDA -I/usr/WS2/corbett5/GEOSX/src/coreComponents -I/usr/WS2/corbett5/GEOSX/src/externalComponents -Iinclude -I/usr/WS2/corbett5/GEOSX/src/coreComponents/linearAlgebra -isystem=/usr/tce/packages/cuda/cuda-10.1.243/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include -isystem=/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/caliper/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/chai/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/raja/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/pugixml/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/conduit/include/conduit -isystem=/usr/WS1/GEOS/GEOSX/TPLs_2021-03-01/install-lassen-clang@upstream-release/hdf5/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/suitesparse/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/superlu_dist/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/trilinos/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/parmetis/include -isystem=/usr/tcetmp/packages/essl/essl-6.2.1/include -isystem=/usr/gapps/GEOSX/thirdPartyLibs/2021-03-01/install-lassen-clang@upstream-release/metis/include -restrict -arch sm_70 --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -g -G -O0 -Xcompiler -O0 -Xcompiler=-fPIC   -Xcompiler=-fopenmp=libomp -std=c++14 -x cu -M /usr/WS2/corbett5/GEOSX/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp -MT coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/CompositionalMultiphaseBase.cpp.o -o coreComponents/physicsSolvers/CMakeFiles/physicsSolvers.dir/fluidFlow/CompositionalMultiphaseBase.cpp.o.d
ptxas fatal   : Cannot take address of function '_ZNK5geosx12constitutive33MultiFluidPVTPackageWrapperUpdate7computeEddRKN7LvArray10ArraySliceIKdLi1ELi0ElEERKNS3_IdLi1ELi0ElEESA_SA_SA_RKNS3_IdLi2ELi1ElEERd' 
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by francoishamon**
This PR moves the implementation of the two `MultiFluidPVTPackageWrapper::compute` functions to `MultiFluidPVTPackageWrapper.hpp` in order to fix #1366.

Resolves #1366

### CODE REVIEW

File: src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp [hunk 1]
```diff
@@ -21,17 +21,11 @@
...
-class MultiphaseSystem;
-
-enum class PHASE_TYPE : int;
-}
-
+#include "constitutive/fluid/MultiFluidUtils.hpp"
 #include <memory>
+// For a reason that I don't understand, adding this include
+// requires adding PVTPackage as a dependent of the physicsSolvers
+#include "pvt/pvt.hpp"
```
Comment (francoishamon) [inline src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp:28]: Moving the two `compute` functions to the hpp file requires including `pvt/pvt.hpp` in `MultiFluidPVTPackageWrapper.hpp`. If I don't add `PVTPackage` as a dependency of `physicsSolvers`, then I get the error: 
```
MultiFluidPVTPackageWrapper.hpp:25:10: fatal error: 'pvt/pvt.hpp' file not found
```
Apparently someone had run into the same issue before (see the deleted comment on GitHub). I spent some time trying to fix this but could not find a good solution. If someone knows how to make this works without making `PVTPackage` a dependency of the `physicsSolvers`, let me know.

Comment (corbett5) [reply src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp:28]: @bmhan12 this isn't going to make your life any easier. Sorry.

**Review by corbett5 — APPROVED** [review]

File: src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp (hunk 1 above)
Comment (rrsettgast) [reply src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp:28]: Can't `PVTPackage` be a dependency of `constitutive`?

Comment (francoishamon) [reply src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp:28]: `PVTPackage` was already a dependency of `constitutive`:
https://github.com/GEOSX/GEOSX/blob/848d9e5e180d0bb90309704c4f8b300daa2c10be/src/coreComponents/constitutive/CMakeLists.txt#L119-L133
and this was enough when `pvt/pvt.hpp` was included in `MultiFluidPVTPackageWrapper.cpp`. Now that `pvt/pvt.hpp` has to be included in  `MultiFluidPVTPackageWrapper.hpp`, it is not enough anymore, and the included file `pvt/pvt.hpp` cannot be found (unless `PVTPackage` is a also dependency of `physicsSolvers`).

Comment (klevzoff) [reply src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp:28]: > Can't `PVTPackage` be a dependency of `constitutive`?

It already is, but `constitutive` isn't a dependency of `physicsSolvers`:
https://github.com/GEOSX/GEOSX/blob/848d9e5e180d0bb90309704c4f8b300daa2c10be/src/coreComponents/physicsSolvers/CMakeLists.txt#L111-L115

Comment (francoishamon) [reply src/coreComponents/constitutive/fluid/MultiFluidPVTPackageWrapper.hpp]: In the last commit I made `physicsSolvers` depend on `constitutive` (and as before, `constitutive` itselft depends on `PVTPackage`).

**Pull request merged**